flink-ml-lib/src/test/java/org/apache/flink/ml/feature/StandardScalerTest.java

```diff
@@ -0,0 +1,279 @@
+                    Vectors.dense(0.69576368, -0.2773501, -1.1547005));
+
+    private final double[] expectedMean = new double[] {0.3, 1, 0};
+    private final double[] expectedStd = new double[] {2.4433583, 7.2111026, 1.7320508};
+    private static final double TOLERANCE = 1e-7;
+
+    @Before
+    public void before() {
+        Configuration config = new Configuration();
+        config.set(ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH, true);
+        env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+        env.setParallelism(4);
+        env.enableCheckpointing(100);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        tEnv = StreamTableEnvironment.create(env);
+        denseTable = tEnv.fromDataStream(env.fromCollection(denseInput)).as("features");
+    }
+
+    @SuppressWarnings("unchecked")
+    private void verifyPredictionResult(
+            List<DenseVector> expectedOutput, Table output, String predictionCol) throws Exception {
+        List<Row> collectedResult =
+                IteratorUtils.toList(tEnv.toDataStream(output).executeAndCollect());
+        List<DenseVector> predictions = new ArrayList<>(collectedResult.size());
+
+        for (Row r : collectedResult) {
+            Vector vec = (Vector) r.getField(predictionCol);
+            predictions.add(vec.toDense());
+        }
+
+        assertEquals(expectedOutput.size(), predictions.size());
+
+        predictions.sort(
+                (vec1, vec2) -> {
+                    int size = Math.min(vec1.size(), vec2.size());
+                    for (int i = 0; i < size; i++) {
+                        int cmp = Double.compare(vec1.get(i), vec2.get(i));
+                        if (cmp != 0) {
+                            return cmp;
+                        }
+                    }
+                    return 0;
+                });
+
+        for (int i = 0; i < predictions.size(); i++) {
+            assertArrayEquals(expectedOutput.get(i).values, predictions.get(i).values, TOLERANCE);
+        }
+    }
+
+    @Test
+    public void testParam() {
+        StandardScaler standardScaler = new StandardScaler();
+
+        assertEquals("features", standardScaler.getFeaturesCol());
+        assertEquals(false, standardScaler.getWithMean());
+        assertEquals(true, standardScaler.getWithStd());
+        assertEquals("prediction", standardScaler.getPredictionCol());
+
+        standardScaler
+                .setFeaturesCol("test_features")
+                .setWithMean(true)
+                .setWithStd(false)
+                .setPredictionCol("test_prediction");
+
+        assertEquals("test_features", standardScaler.getFeaturesCol());
+        assertEquals(true, standardScaler.getWithMean());
+        assertEquals(false, standardScaler.getWithStd());
+        assertEquals("test_prediction", standardScaler.getPredictionCol());
+    }
+
+    @Test
+    public void testFeaturePredictionParam() {
```

Comment:
nits: suppose we change the parameter to inputCol and outputCol, it might not be intuitive to name this method as `testFeaturePredictionParam`. How about changing it to `testOutputSchema()`? We can use this name consistently across other tests.